### PR TITLE
DAOS-15774 test: fix nvme/enospace.py container destroy

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: NvmeEnospace
Skip-unit-tests: true
Skip-fault-injection-test: true

Track containers created with start_ior_load and delete them explicitly.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
Test passed with skipping register_cleanup:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14285/4/testReport/

File: src/tests/ftest/util/test_utils_container.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def add_container(test, pool, namespace=CONT_NAMESPACE, create=True, daos=None,
         container.update_params(**params)
     if create:
         container.create()
-    test.register_cleanup(remove_container, test=test, container=container)
+    if container.register_cleanup.value is True:
```
Comment (phender) [inline src/tests/ftest/util/test_utils_container.py:46]: I don't think we can properly identify a TestContainer object as not needing to be cleaned up when we instantiate the object.  Some other action needs to be performed - like destroying its parent pool - before we know TestContainer.destroy() should be skipped.  If we:

1. Create a pool
2. Create a container
3. Run ior
4. Destroy the pool
5. Perform additional test steps
6. Enter tearDown()

Having tearDown() call TestContainer.destroy() is valid if the test times out any time after step #2 and before step #4.  In this case we need to notify TestContainer that it should ignore any calls to destroy() either just before step #4 or after it completes successfully.

Comment (daltonbohning) [reply src/tests/ftest/util/test_utils_container.py:46]: If we destroy the pool, all containers in that pool are also destroyed. We don't necessarily need to destroy each container individually for every test case.
And this particular test is doing this to destroy containers:
https://github.com/daos-stack/daos/blob/c7675eb0c55745b3d19b974e44574dcc87211044/src/tests/ftest/nvme/enospace.py#L206

The problem is we don't have a reference to the container after it's created. And the "proper" fix is to refactor the utilities.

There are also generally cases where a test creates A LOT of containers and we don't need to destroy each container individually - it's enough to just destroy the pool. Like this one which creates 100 containers and is also broken right now:
https://github.com/daos-stack/daos/blob/c7675eb0c55745b3d19b974e44574dcc87211044/src/tests/ftest/nvme/pool_capacity.py#L224

Comment (daltonbohning) [reply src/tests/ftest/util/test_utils_container.py:46]: If the pool destroy has already been registered, it's not necessary to register container destroy and then just call skip_cleanup. IMO having to call `container.skip_cleanup` is more of a hack than not registering the cleanup in the first place

**Review by phender — APPROVED** [review]

**Comment by daltonbohning** [comment]
Modified test passed, and arbitrarily IorSmall for shared utility verification:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14285/8/testReport/

**Review by dinghwah — APPROVED** [review]

**Pull request merged**